### PR TITLE
Test all valid immediates in `_mm_dp_pd`

### PR DESCRIPTION
**Pull request by Cuda-Chen**
Test all valid immediates in `_mm_dp_pd`.

### CODE REVIEW

File: tests/impl.cpp [hunk 1]
```diff
@@ -8267,25 +8267,46 @@ result_t test_mm_cvtepu8_epi64(const SSE2NEONTestImpl &impl, uint32_t iter)
...
+        for (size_t i = 0; i < 2; i++)                               \
+            sum += ((imm) & (1 << (i + 4))) ? _a[i] * _b[i] : 0;     \
+        for (size_t i = 0; i < 2; i++)                               \
+            d[i] = (imm & (1 << i)) ? sum : 0;                       \
+        __m128d a = load_m128d(_a);                                  \
+        __m128d b = load_m128d(_b);                                  \
+        __m128d ret = _mm_dp_pd(a, b, imm);                          \
+        if (validateDouble(ret, d[0], d[1]) != TEST_SUCCESS)         \
+            return TEST_FAIL;                                        \
+    } while (0);
```
Comment (jserv) [inline tests/impl.cpp:8286]: Remove the trailing semicolon (`;`) and append it at the end of each occurrence of `ENUM_MM_DP_PD_TEST_CASES`.

Comment (Cuda-Chen) [reply tests/impl.cpp]: Done.

File: tests/impl.cpp [hunk 2]
```diff
@@ -8267,25 +8267,46 @@ result_t test_mm_cvtepu8_epi64(const SSE2NEONTestImpl &impl, uint32_t iter)
     return validateInt64(ret, i0, i1);
 }
 
+#define ENUM_MM_DP_PD_TEST_CASES(imm8)                               \
```
Comment (jserv) [inline tests/impl.cpp:8270]: Rename to `MM_DP_PD_TEST_CASE_WITH`.

Comment (Cuda-Chen) [reply tests/impl.cpp]: Done.

**Comment by jserv** [comment]
Thank @Cuda-Chen for contributing!

**Pull request merged**